src/main/utils/resource.ts

```diff
@@ -0,0 +1,155 @@
+    }
+
+    /**
+     * Download the resource to the given folder
+     * @param downloadToFolder - the folder that the resource will be downloaded to
+     * @returns the full path of the file that was downloaded successfully, undefined otherwise
+     */
+    private async downloadToFolder(downloadToFolder: string = this._targetDir): Promise<string> {
+        let resourcePath: string = path.join(downloadToFolder, Utils.getLastSegment(this.sourceUrl));
+        // Download new
+        await this._connectionManager
+            .artifactory()
+            .download()
+            .downloadArtifactToFile(this.sourceUrl, resourcePath);
+        return resourcePath;
+    }
+
+    private removeOldTargetAndCopyFile(filePath: string) {
+        if (this.isExists()) {
+            fs.rmSync(this._targetPath);
+        }
+        fs.copyFileSync(filePath, this._targetPath);
+        fs.chmodSync(this._targetPath, this._mode);
+    }
+
+    private removeOldTargetDirAndExtractZip(filePath: string) {
+        if (fs.existsSync(this._targetDir)) {
+            ScanUtils.removeFolder(this._targetDir);
+        }
+        ScanUtils.extractZip(filePath, this._targetDir);
+    }
+
+    /**
+     * if the given temp file is a binary within a zip, replace and extract its content to the target dir.
+     * if the given file is a binary, replace and copy it to the target path
+     * @param tempPath - the file to copy into target
+     */
+    public copyToTarget(tempPath: string) {
+        if (tempPath.endsWith('.zip')) {
+            this.removeOldTargetDirAndExtractZip(tempPath);
+        } else {
+            this.removeOldTargetAndCopyFile(tempPath);
+        }
+    }
+
+    /**
+     * Update the resource to the latest version.
+     * if remoteSha256 is not set a request for the remote will be sent to check the most recent checksum
+     * @returns true if the resource was updated, false otherwise
+     */
+    public async update(): Promise<boolean> {
+        let tmpFolder: string = ScanUtils.createTmpDir();
+        try {
+            this._logManager.logMessage('Starting to update resource ' + this._name + ' from ' + this.sourceUrl, 'DEBUG');
+            let resourceTmpPath: string = await this.downloadToFolder(tmpFolder);
+            if (this.calculateLocalChecksum(resourceTmpPath) !== (this._cacheRemoteSha256 ?? (await this.calculateRemoteChecksum()))) {
+                this._logManager.logMessage('Resource ' + this._name + ' update failed.', 'ERR');
+                return false;
+            }
+            this.copyToTarget(resourceTmpPath);
+            this._logManager.logMessage('Resource ' + this._name + ' was update successfully.', 'DEBUG');
+            return true;
+        } finally {
+            ScanUtils.removeFolder(tmpFolder);
+        }
+    }
+
+    /**
+     * Check if the resource exists in the provided target path.
+     * @returns true if file exists in the path, false otherwise
+     */
```

Comment:
A comment like this can sometimes be useful, but the function name conveys the information better.